velox/exec/RankLikeWindowBuild.h

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include "velox/exec/WindowBuild.h"
+
+namespace facebook::velox::exec {
+
+/// If the data has already been sorted according to the partition key and sort
+/// key, there is no need to hold all the data of a partition in memory for Rank
```

Comment:
Could we add more description for this class on how the optimization is achieved? And do we need to update the documentation https://github.com/facebookincubator/velox/blob/main/velox/docs/develop/window.rst?